src/libxtp/tools/mol2orb.cc

```diff
@@ -0,0 +1,272 @@
+    // energy line
+    std::getline(input_file, line);
+    iss.str(line);
+    iss.clear();
+    iss >> tempStr >> tempDouble;
+    orbitals.MOs().eigenvalues()[i] = tempDouble;
+    // spin channel line
+    std::getline(input_file, line);
+    iss.str(line);
+    iss.clear();
+    iss >> tempStr >> tempStr;
+    if (tempStr == "Beta") {
+      XTP_LOG(Log::error, _log)
+          << "Open shell systems are currently not supported" << std::flush;
+    }
+    // occupation line
+    std::getline(input_file, line);
+    iss.str(line);
+    iss.clear();
+    iss >> tempStr >> tempDouble;
+    number_of_electrons += (int)tempDouble;
+
+    // MO coefficients
+    for (int j = 0; j < basis_size; j++) {  // loop over ao's
+      std::getline(input_file, line);
+      iss.str(line);
+      iss.clear();
+      iss >> tempInt >> tempDouble;
+      orbitals.MOs().eigenvectors()(j, i) = tempDouble;
+    }
+  }
+
+  orbitals.setNumberOfAlphaElectrons(number_of_electrons);
+  orbitals.setNumberOfOccupiedLevels(number_of_electrons / 2);
+
+  // Get the correct ordering of atomic orbitals
+  // Note that the molden specification specifies its own orbital ordering
+  // Hence we can't use the qmpackage's ordering.
+  reorderOrbitals(orbitals.MOs().eigenvectors());
+
+  getline(input_file, line);
+  return line;
+}
+
+bool Mol2Orb::Evaluate() {
+  _log.setReportLevel(Log::current_level);
+  _log.setMultithreading(true);
+  _log.setCommonPreface("\n... ...");
+
+  Orbitals orbitals;
+
+  // Open the file
+  std::ifstream input_file(_moldenfile);
+  // Check if succesfull
+  if (input_file.fail()) {
+    XTP_LOG(Log::error, _log) << "Could not open " << _moldenfile << std::flush;
+    return false;
+  }
+
+  XTP_LOG(Log::error, _log)
+      << "Started parsing " << _moldenfile << " to " << _orbfile << std::flush;
+
+  QMMolecule& mol = orbitals.QMAtoms();
+
+  std::string line;
+  std::getline(input_file, line);
+  while (input_file) {
+    boost::trim(line);
+    if (line[0] != '[') {
+      // ignore non-relevant lines
+      std::getline(input_file, line);
+      continue;
+    }
+
+    // Extract the part between square brackets
+    long unsigned close = line.find("]");
+    std::string sectionType = line.substr(1, close - 1);
+
+    // Import data from relevant sections
+    if (sectionType == "Atoms") {
+      std::string units = line.substr(close + 1);  // extra units Atomic Units
+                                                   // (AU) or Angstrom (Angs)
+      boost::trim(units);
+      XTP_LOG(Log::error, _log)
+          << "Reading atoms using " << units << " units." << std::flush;
+      line = readAtoms(mol, units, input_file);
+      setupBasis(orbitals);
+    } else if (sectionType == "GTO") {
+      XTP_LOG(Log::error, _log)
+          << "Basisset specification is ignored." << std::flush;
+      XTP_LOG(Log::error, _log)
+          << "Basissets are specified via the mol2orb.xml options file."
```

Comment:
why not read the basisset from molden as well? What are the dangers of doing it like this?